src/Processors/Transforms/Streaming/AggregatingTransform.cpp

```diff
@@ -244,11 +244,25 @@ void AggregatingTransform::setCurrentChunk(Chunk chunk, const ChunkContextPtr &
 
         current_chunk_aggregated.setChunkContext(std::move(chunk_ctx));
     }
+
+    if (retracted_chunk.rows())
+    {
+        current_chunk_retracted = std::move(retracted_chunk);
+        current_chunk_retracted.getOrCreateChunkContext()->setRetractedDataFlag();
+    }
 }
 
 IProcessor::Status AggregatingTransform::preparePushToOutput()
 {
     auto & output = outputs.front();
+
+    /// At first, push retracted data, then push aggregated data
+    if (current_chunk_retracted)
+    {
```

Comment:
Given there are two pushes: one for retracted chunk, the other for aggregated chunk, is it possible the end user got incorrect intermediate result which only include retracted chunk but without aggregated chunk?

Reply:
No, the retracted chunk with flag `RETRACTED_DATA_FLAG`, so don't emit intermediate result

Reply:
I only saw JoinTransform handle this **RETRACTED_DATA_FLAG**. How about the aggregation without join? Is it possible the end user get intermediate result?

Reply:
`RETRACTED_DATA_FLAG` data don't have watermark, since `WatermarkTransform` will skip them.